ocaml/xapi/xapi_vm_appliance_lifecycle.ml

```diff
@@ -37,6 +37,8 @@ let check_operation_error ~__context record self op =
         (* Can hard_shutdown/shutdown if any are not halted. *)
         | `hard_shutdown | `shutdown ->
           (fun power_state -> power_state <> `Halted), "All VMs in this appliance are halted."
+        | `unknown ->
+          (fun _ -> true), "Unknown power state"
```

Comment:
This makes me wonder about using "unknown" rather than something like "unrecognised" for this whole pull-request.
"Unknown power state" sounds as if we don't know the state: as if it could be running or suspended or whatever.
In fact we know that the state is not running, halted or any of the other states known to us: we know it's something else.

Reply:
..so you're saying it's unknown to us :-)

I'm going to leave it as it is I think.